src/Perun/Offchain.hs

```diff
@@ -209,12 +206,13 @@ fund FundParams {..} = do
   --      try to use fixed-sized arrays instead
   logInfo @P.String $ printf "Funded %d Lovelace for party %d on channel %d" (balances state !! fpIndex) fpIndex fpChannelId
 
-abort :: AbortParams -> Contract w s Text ()
+abort :: AbortParams -> Contract w s PerunError ()
 abort (AbortParams cId) = do
   (oref, o, d@ChannelDatum {..}) <- findChannel cId
   logInfo @P.String $ printf "found channel utxo with datum %s" (P.show d)
-  unless (all isLegalOutValue funding) . throwError . pack $ printf "Unable to abort channel with any funding below minimum Ada"
-  when funded . throwError . pack $ printf "can not abort funded channel"
+  -- TODO: This still gives the potential to lock a channel?
+  unless (all isLegalOutValue funding) . throwError $ InsufficientMinimumAdaFundingError
+  when funded . throwError $ AlreadyFundedAbortError
```

Comment:
I don't think it does, because we check on-chain on `Fund` calls that in the new state, every balance is valid (i.e. **not in** `(0, minAda)` ). This means there can be no channel that has any illegal out-value unless right after calling start (because it does not execute any on-chain script), in which the other party should verify/detect this and just *not fund the channel*. 
Generally, other parties have to verify the channel after it is started by one party, because this party could just start it with `any` parameters in the datum. The other parties would react to this with just not interacting with this channel, which is fine because only the party that maliciously started the channel would have locked their funds at this point.

So if you see no other way the channel might be locked here, this TODO can be removed.

Reply:
Huch, I accidentally left this comment there. Thanks for bringing it up and reassuring, I also think that this `TODO` is of no relevance anymore. (It was a _sidethought_ when I refactored the error messages)